ros/lib/utils.py

```diff
@@ -80,20 +78,22 @@ def user_data_from_identity(identity):
     return identity['user']
 
 
-def validate_type(value, type_):
+def validate_system(reporter, msg):
     """
-    Validate the type of a value.
-    Currently available types: bool
-    :param value: Value to validate.
-    :param type_: Type to validate against.
-    :return: True if the value is of the specified type, False otherwise.
+    Validate msg from consumers.
+    :param reporter: consumer reporter
+    :param msg: msg to validate.
+    :return: True if the msg is from a cloud provider and is_ros flag is set to true, False otherwise.
     """
-    if type_ == bool:
-        # ast.literal_eval does not understand lowercase 'True' or 'False'
-        value = value.capitalize() if value in ['true', 'false'] else value
-    evaluated_value = type_evaluation.literal_eval(value) if value else None
-
-    return True if type(evaluated_value) == type_ else False
+    is_ros = False
+    cloud_provider = None
+    if reporter == 'INVENTORY EVENTS':
+        is_ros = msg["platform_metadata"].get("is_ros")
+        cloud_provider = msg['host']['system_profile'].get('cloud_provider')
+    else:
+        is_ros = msg["input"]["platform_metadata"].get("is_ros")
+        cloud_provider = msg["results"]["system"]["metadata"].get('cloud_provider')
```

Comment:
Do you think we should capture info which helps in getting how many requests ros backend gets for non-cloud?

Reply:
> how many requests ros backend gets for non-cloud?

IMO that is for future scope. It would be good to keep it simple as those logs would unnecessarily fill up our logs.

Reply:
Do you mean log statement or actually something app or org level to keep track of this? I think if we can keep track of org level maybe just using some var with log then also it will help as insight? However it wont persist on app reboot, maybe some table also to persist it just as count?

Reply:
> Do you mean log statement or actually something app or org level to keep track of this? I think if we can keep track of org level maybe just using some var with log then also it will help as insight? However it wont persist on app reboot, maybe some table also to persist it just as count?

yes, I would prefer to have short log statement.

Reply:
> > I also discussed about this with the @r14chandra, and below are our thoughts,
> > ```
> > 1. We can have logs however those logs will only help for insights, and then Kibana instance is not that good when it comes to relating the logs from different services. I mean, something like, can I get total no of non cloud and is_ros systems when compared to central truth HBI? maybe not.
> > ```
> 
> Why we want to related logs from different services? Also, intention here is not to get total number of non-cloud systems vs is_ros systems.
> 
Yeah I was thinking to count total number of systems based on HBI.
> > ```
> > 2. The logs won't provide any advantage for troubleshooting? because we don't support this use case and even if customer complains why system is not in ROS then support team can actually check the facts in sosreport and communicate it.
> > ```
> 
> Definitely, this log statement will not be for troubleshooting.
> 
> > ```
> > 3. And yes we will fill up the kibana logs even when its not that useful.
> > ```
> 
> Could you suggest any alternative to get how many requests come to ROS backend where is_ros is true but not AWS instance? Though it's not useful for troubleshooting, we kept it for a use case. I guess you can confirm with Pau and decide to skip.

If there is use case then for sure we should add the logging, @pgarciaq request to confirm same.